render_bundle/applications.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Classes describing the content of each of the applications used in bundles."""
+
+from lib.charm_bundle_generator import Application, Resource
+
+
+class UPF(Application):
```

Comment:
UPF operator also supports a number of config options. Perhaps it would be a good idea to model those in a `charm_bundle_generator`? Or should we postpone that until it's absolutely necessary?